NHV-117 - Using Validator and Envers together, fix & test

AlexanderKot opened the pull request:
[NHV-117](https://nhibernate.jira.com/browse/NHV-117) - Using Validator and Envers together

fredericDelaporte commented on src/NHibernate.Validator/Event/ValidatePreInsertEventListener.cs:
```diff
@@ -46,7 +46,7 @@ public void Initialize(Configuration cfg)
 			}
 
-			IEnumerable<PersistentClass> classes = cfg.ClassMappings.Where(clazz => clazz.ClassName != null).Where(clazz => !clazz.ClassName.StartsWith("NHibernate.Envers"));
+			IEnumerable<PersistentClass> classes = cfg.ClassMappings.Where(clazz => clazz.ClassName != null && clazz.MappedClass != null);
```
I would tend to change that to `cfg.ClassMappings.Where(clazz => clazz.HasPocoRepresentation);`

Testing both `ClassName` and `MappedClass` is redundant, `MappedClass` reflect `ClassName`. (And `HasPocoRepresentation` is just `ClassName != null`, but with a more meaningful naming.)

The whole old test was added in 6449be36d129881045e0e0527ce3ac97b38ff1b5, I would like to know why filtering manually envers classes was considered required. If not filtering it may cause failure, we need a test for that. @TimurZanagar, can you tell us?

AlexanderKot commented:
OK about PR\commit names and one-liner.

I really do not remember have I changed ValidatorInitializer.cs before or after reading NHV-117 comments, it was done locally ~ 2 years ago. But solution was the same.
I have just create test yesterday and changed ValidatePreInsertEventListener.cs to be similar with previous file and do not have Envers name hard-coded as string.

Checking by persistentClass.MappedClass works successfully in my project for this years.
Tests are also green with  clazz.HasPocoRepresentation. 
As you know NHib internals better - it is up to you which variant to use.
So I could commit changes to ...HasPocoRepresentation.

fredericDelaporte commented:
I have made the changes.
